core/src/main/java/jenkins/triggers/ReverseBuildTrigger.java

```diff
@@ -113,7 +114,12 @@ private boolean shouldTrigger(Run upstreamBuild, TaskListener listener) {
             return false;
         }
         // This checks Item.READ also on parent folders; note we are checking as the upstream auth currently:
-        boolean downstreamVisible = jenkins.getItemByFullName(job.getFullName()) == job;
+        boolean downstreamVisible = false;
+        try {
+            downstreamVisible = jenkins.getItemByFullName(job.getFullName()) == job;
+        } catch (AccessDeniedException ex) {
+            // ignore, we will fall-back later
```

Comment:
downstream is discoverable (visible?), but not readable. There is some terminology mess, which IMHO leads to incorrect messages below in this case.